internal/execution/execution.go

```diff
@@ -114,9 +115,17 @@ func call(ctx context.Context, url string, timeout time.Duration, body []byte) (
 	}
 	defer resp.Body.Close()
 
+	return handleResponse(resp)
+}
+
+func handleResponse(resp *http.Response) ([]byte, error) {
+	data, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
 	// Check for success between 200 and 299, redirect 300 to 399 is handled by the client, return error with statusCode >= 400
 	if resp.StatusCode >= 200 && resp.StatusCode <= 299 {
-		return io.ReadAll(resp.Body)
+		return data, nil
 	}
-	return nil, zerrors.ThrowUnknown(nil, "EXEC-dra6yamk98", "Errors.Execution.Failed")
+	return nil, zerrors.ThrowUnknown(fmt.Errorf("StatusCode: %d, Response: \"%s\"", resp.StatusCode, string(data)), "EXEC-dra6yamk98", "Errors.Execution.Failed")
```

Comment:
This would not allow clients to parse the error easily. 

additionally we know the status code. why do we return an unkonwn error?

Reply:
And i just saw that the tests don't check for the correct error body

Reply:
Good point, I created a small mapper, which knows the same StatusCodes as handled from ZERRORS to http.StatusCodes. Still not quite sure about if this error should still be wrapped, as the error is the reason, but does not necessarily tell you how ZITADEL handled it.